benchexec/tools/legion.py

```diff
@@ -27,11 +27,20 @@ class Tool(benchexec.tools.template.BaseTool):
         "lib",
     ]
 
-    def executable(self):
-        return util.find_executable("legion-sv")
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("legion-sv")
 
     def version(self, executable):
         return self._version_from_tool(executable, ignore_stderr=False)
 
     def name(self):
         return "Legion"
+
+    def cmdline(self, executable, options, task, rlimits):
+        data_model_param = get_data_model_from_task(
+            task, {"ILP32": "-32", "LP64": "-64"}
+        )
+        if data_model_param and data_model_param not in options:
+            options += [data_model_param]
+
+        return super().cmdline(executable, options, task, rlimits)
```

Comment:
I would suggest to simply return the value here instead of calling `super`. Having the full picture of what is used as command line in one method seems worth more than saving one line of code.

Reply:
fixed in 4b7464cc40a620880e02a86218a0eb1d0b26ac8c